battle-engine.js

```diff
@@ -1052,11 +1052,9 @@ BattlePokemon = (function () {
 		if (!sourceEffect && this.battle.effect) sourceEffect = this.battle.effect;
 		if (!source && this.battle.event && this.battle.event.target) source = this.battle.event.target;
 		item = this.getItem();
-		if (this.battle.runEvent('UseItem', this, null, null, item) && this.battle.runEvent('EatItem', this, null, null, item)) {
```

Comment:
Continue working on this. The person assigned to a pull request is the one who reviews it before merging (usually), and Marty knows the most about game mechanics.